data/beaconrestapi/src/main/java/tech/pegasys/teku/beaconrestapi/BeaconRestApiConfig.java

```diff
@@ -118,7 +124,20 @@ public int getMaxUrlLength() {
   }
 
   public int getValidatorThreads() {
-    return validatorThreads;
+    if (validatorThreads.isPresent()) {
+      return validatorThreads.get();
+    }
+
+    final int coreNumbers = Runtime.getRuntime().availableProcessors();
```

Comment:
maybe rename this as `availableProcessors`, `coreNumbers` is kinda hard to understand, i'd settle for `numberOfCores` or similar...